python/cudf/cudf/_fuzz_testing/utils.py

```diff
@@ -312,13 +312,8 @@ def sanitize(value, struct_type):
         return tuple(values_list)
 
     has_nulls_or_nullable_dtype = any(
-        [
-            True
-            if df[col].dtype in pandas_dtypes_to_np_dtypes
-            or df[col].isnull().any()
-            else False
-            for col in df.columns
-        ]
+        df[col].dtype in pandas_dtypes_to_np_dtypes or df[col].isnull().any()
+        for col in df.columns
```

Comment:
There's no walrus in the snippet. Did you mean `(col := df[colname]).dtype`?

Reply:
@shwina I don't know when it's safe to use `df._data.columns` or not. Always? Sometimes?

Reply:
I chose not to use `df._data` in this case because fuzz testing utilities seem like a place to rely on public APIs, not internal functions. For internal usage it is generally safe to loop over the columns of `df._data` instead of `df`, but it is not always safe to try to access a single column if that column name comes from the user. The reason is that user-provided columns (e.g. via `df['my_col_name']`) can a) require additional parsing that is done by `DataFrame.__getitem__`, or b) the `DataFrame` may have a MultiIndex for its columns.

Reply:
Okay. At first it looked like the fuzz testing was internal, but I see no other internal APIs are being called there. I'll revert to the walrus.

Reply:
Sorry for the runaround. I don't think it's a big deal either way, I was just explaining my rationale. The discussion on this usage of the walrus was the important piece for me, not the actual change. The performance of this particular line of code is probably irrelevant anyway, so I'd leave the code in whatever state it is now and call it good 🙂.